### PR TITLE
Updated tests for Marketplace new UI look, Desktop and Mobile

### PR DESCRIPTION
**Pull request by AndreiH**
Seems like Marketplace is going to change the UI.
This changes only apply to DEV and STAGE for now.

### CODE REVIEW

File: tests/desktop/consumer_pages/test_details_page.py [hunk 1]
```diff
@@ -73,7 +73,6 @@ def test_that_reports_abuse_as_anonymous_user(self, mozwebqa):
         report_abuse_box = details_page.click_report_abuse_button()
 
         Assert.true(report_abuse_box.is_visible)
-        Assert.false(report_abuse_box.is_report_button_enabled)
```
Comment (AndreiH) [inline tests/desktop/consumer_pages/test_details_page.py]: The button is always enabled, we just get an error message if we don't enter any text in the box

**Comment by bobsilverberg** [comment]
@AndreiH Have you set up an adhoc for any of these? I have run some locally and am seeing failures, but wonder if the same would happen on CI

**Comment by bobsilverberg** [comment]
@AndreiH This is looking pretty good. There are a four tests that are failing for me locally though. 

test_opening_every_category_page_from_categories_menu 
test_settings_dropdown_menu
test_that_checks_the_editing_of_a_review
test_results_page_items

Stack traces below:

```
================================================================= FAILURES ==================================================================
__________________________________ TestConsumerPage.test_opening_every_category_page_from_categories_menu ___________________________________

self = <tests.desktop.consumer_pages.test_home_page.TestConsumerPage instance at 0x1083a63f8>
mozwebqa = <pytest_mozwebqa.pytest_mozwebqa.TestSetup instance at 0x1083a6680>

    @pytest.mark.sanity
    @pytest.mark.nondestructive
    def test_opening_every_category_page_from_categories_menu(self, mozwebqa):

        home_page = Home(mozwebqa)
        home_page.go_to_homepage()

        for i in range(home_page.category_count):
            home_page.hover_over_categories_menu()
            category_name = home_page.categories.items[i].name
            category_page = home_page.categories.items[i].click_category()
            Assert.equal(category_name.title(), category_page.category_title)
            Assert.true(category_page.is_the_current_page)
            Assert.true(len(category_page.apps) > 0)
>           Assert.true(category_page.is_new_popular_tabs_visible)

tests/desktop/consumer_pages/test_home_page.py:66:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <class 'unittestzero.Assert'>, first = False, msg = None

    @classmethod
    def true(self, first, msg=None):
        """
            Asserts that what is given is equal to True

            :Args:
             - First object to be tested
             - Message that will be printed if it fails
            """

>       assert bool(first) is True, msg
E       AssertionError: None

../../.virtualenvs/marketplace-tests/lib/python2.7/site-packages/unittestzero.py:79: AssertionError

_______________________________________________ TestConsumerPage.test_settings_dropdown_menu ________________________________________________

self = <tests.desktop.consumer_pages.test_home_page.TestConsumerPage instance at 0x1053e63f8>
mozwebqa = <pytest_mozwebqa.pytest_mozwebqa.TestSetup instance at 0x1053e6680>

    @pytest.mark.sanity
    @pytest.mark.nondestructive
    def test_settings_dropdown_menu(self, mozwebqa):
        home_page = Home(mozwebqa)
        home_page.go_to_homepage()

        home_page.header.click_sign_in()
        home_page.login(user="default")

        # Verify account settings menu
        user_settings = home_page.header.click_edit_account_settings()
        Assert.true(user_settings.is_the_current_page)
        Assert.true(user_settings.is_email_visible)
        Assert.true(user_settings.is_display_name_visible)
        Assert.true(user_settings.is_region_field_visible)
        Assert.true(user_settings.is_save_button_visible)
        Assert.true(user_settings.is_sign_out_button_visible)

        # Verify My Apps menu
        home_page.go_to_homepage()
        my_apps_page = home_page.header.click_my_apps()
        Assert.true(my_apps_page.is_the_current_page)
>       my_apps_page.click_expand_button()

tests/desktop/consumer_pages/test_home_page.py:114:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pages.desktop.consumer_pages.account_settings.My_Apps object at 0x1053e7950>

    def click_expand_button(self):
>       self.find_element(*self._expand_button_locator).click()

pages/desktop/consumer_pages/account_settings.py:129:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pages.desktop.consumer_pages.account_settings.My_Apps object at 0x1053e7950>
locator = ('css selector', '.app-list-filters .expand-toggle')

    def find_element(self, *locator):
>       return self._selenium_root.find_element(*locator)

pages/page.py:137:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.firefox.webdriver.WebDriver object at 0x105444b50>, by = 'css selector'
value = '.app-list-filters .expand-toggle'

    def find_element(self, by=By.ID, value=None):
        """
            'Private' method used by the find_element_by_* methods.

            :Usage:
                Use the corresponding find_element_by_* instead of this.

            :rtype: WebElement
            """
        if not By.is_valid(by) or not isinstance(value, str):
            raise InvalidSelectorException("Invalid locator values passed in")

        return self.execute(Command.FIND_ELEMENT,
>                            {'using': by, 'value': value})['value']

../../.virtualenvs/marketplace-tests/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py:662:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.firefox.webdriver.WebDriver object at 0x105444b50>, driver_command = 'findElement'
params = {'sessionId': u'821a053c-f4ab-6743-8b13-3a677221bdb6', 'using': 'css selector', 'value': '.app-list-filters .expand-toggle'}

    def execute(self, driver_command, params=None):
        """
            Sends a command to be executed by a command.CommandExecutor.

            :Args:
             - driver_command: The name of the command to execute as a string.
             - params: A dictionary of named parameters to send with the command.

            :Returns:
              The command's JSON response loaded into a dictionary object.
            """
        if self.session_id is not None:
            if not params:
                params = {'sessionId': self.session_id}
            elif 'sessionId' not in params:
                params['sessionId'] = self.session_id

        params = self._wrap_value(params)
        response = self.command_executor.execute(driver_command, params)
        if response:
>           self.error_handler.check_response(response)

../../.virtualenvs/marketplace-tests/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py:173:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x1054447d0>
response = {u'name': u'findElement', u'sessionId': u'821a053c-f4ab-6743-8b13-3a677221bdb6', u'status': 7, u'value': {u'message': ...mponents/driver-component.js:548', u'lineNumber': 5, u'methodName': u'fxdriver.Timer.prototype.setTimeout/<.notify'}]}}

    def check_response(self, response):
        """
            Checks that a JSON response from the WebDriver does not have an error.

            :Args:
             - response - The JSON response from the WebDriver server as a dictionary
               object.

            :Raises: If the response contains an error message.
            """
        status = response['status']
        if status == ErrorCode.SUCCESS:
            return
        exception_class = ErrorInResponseException
        if status in ErrorCode.NO_SUCH_ELEMENT:
            exception_class = NoSuchElementException
        elif status in ErrorCode.NO_SUCH_FRAME:
            exception_class = NoSuchFrameException
        elif status in ErrorCode.NO_SUCH_WINDOW:
            exception_class = NoSuchWindowException
        elif status in ErrorCode.STALE_ELEMENT_REFERENCE:
            exception_class = StaleElementReferenceException
        elif status in ErrorCode.ELEMENT_NOT_VISIBLE:
            exception_class = ElementNotVisibleException
        elif status in ErrorCode.INVALID_ELEMENT_STATE:
            exception_class = InvalidElementStateException
        elif status in ErrorCode.INVALID_SELECTOR \
                or status in ErrorCode.INVALID_XPATH_SELECTOR \
                or status in ErrorCode.INVALID_XPATH_SELECTOR_RETURN_TYPER:
            exception_class = InvalidSelectorException
        elif status in ErrorCode.ELEMENT_IS_NOT_SELECTABLE:
            exception_class = ElementNotSelectableException
        elif status in ErrorCode.INVALID_COOKIE_DOMAIN:
            exception_class = WebDriverException
        elif status in ErrorCode.UNABLE_TO_SET_COOKIE:
            exception_class = WebDriverException
        elif status in ErrorCode.TIMEOUT:
            exception_class = TimeoutException
        elif status in ErrorCode.SCRIPT_TIMEOUT:
            exception_class = TimeoutException
        elif status in ErrorCode.UNKNOWN_ERROR:
            exception_class = WebDriverException
        elif status in ErrorCode.UNEXPECTED_ALERT_OPEN:
            exception_class = UnexpectedAlertPresentException
        elif status in ErrorCode.NO_ALERT_OPEN:
            exception_class = NoAlertPresentException
        elif status in ErrorCode.IME_NOT_AVAILABLE:
            exception_class = ImeNotAvailableException
        elif status in ErrorCode.IME_ENGINE_ACTIVATION_FAILED:
            exception_class = ImeActivationFailedException
        elif status in ErrorCode.MOVE_TARGET_OUT_OF_BOUNDS:
            exception_class = MoveTargetOutOfBoundsException
        else:
            exception_class = WebDriverException
        value = response['value']
        if isinstance(value, basestring):
            if exception_class == ErrorInResponseException:
                raise exception_class(response, value)
            raise exception_class(value)
        message = ''
        if 'message' in value:
            message = value['message']

        screen = None
        if 'screen' in value:
            screen = value['screen']

        stacktrace = None
        if 'stackTrace' in value and value['stackTrace']:
            stacktrace = []
            try:
                for frame in value['stackTrace']:
                    line = self._value_or_default(frame, 'lineNumber', '')
                    file = self._value_or_default(frame, 'fileName', '<anonymous>')
                    if line:
                        file = "%s:%s" % (file, line)
                    meth = self._value_or_default(frame, 'methodName', '<anonymous>')
                    if 'className' in frame:
                        meth = "%s.%s" % (frame['className'], meth)
                    msg = "    at %s (%s)"
                    msg = msg % (meth, file)
                    stacktrace.append(msg)
            except TypeError:
                pass
        if exception_class == ErrorInResponseException:
            raise exception_class(response, message)
        elif exception_class == UnexpectedAlertPresentException and 'alert' in value:
            raise exception_class(message, screen, stacktrace, value['alert'].get('text'))
>       raise exception_class(message, screen, stacktrace)
E       NoSuchElementException: Message: Unable to locate element: {"method":"css selector","selector":".app-list-filters .expand-toggle"}
E       Stacktrace:
E           at FirefoxDriver.prototype.findElementInternal_ (file:///var/folders/sk/1kj23lvn46n7pvr7_xk074wm0000gn/T/tmpXrINI5/extensions/fxdriver@googlecode.com/components/driver-component.js:9641:26)
E           at fxdriver.Timer.prototype.setTimeout/<.notify (file:///var/folders/sk/1kj23lvn46n7pvr7_xk074wm0000gn/T/tmpXrINI5/extensions/fxdriver@googlecode.com/components/driver-component.js:548:5)

../../.virtualenvs/marketplace-tests/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py:166: NoSuchElementException

___________________________________________ TestReviews.test_that_checks_the_editing_of_a_review ____________________________________________

self = <tests.desktop.consumer_pages.test_reviews.TestReviews instance at 0x11187c3f8>
mozwebqa = <pytest_mozwebqa.pytest_mozwebqa.TestSetup instance at 0x11187c680>

    @pytest.mark.sanity
    @pytest.mark.credentials
    def test_that_checks_the_editing_of_a_review(self, mozwebqa):

        self._reviews_setup(mozwebqa)

        home_page = Home(mozwebqa)
        home_page.go_to_homepage()

        # Login into Marketplace
        home_page.header.click_sign_in()
        home_page.login(user="default")
        Assert.true(home_page.is_the_current_page)

        # Search for the test app and go to its details page
        search_page = home_page.header.search(self.app_name)
        details_page = search_page.results[0].click_name()
        Assert.true(details_page.is_the_current_page)

        details_page.wait_for_review_button_visible()
        Assert.equal(details_page.review_button_text, "Edit your review")

        # Write a review
        edit_review = details_page.click_review_button(edit_review=True)
        mock_review = MockReview()
        details_page = edit_review.write_a_review(mock_review['rating'], mock_review['body'])

        # Check notification
        details_page.wait_notification_box_visible()
        Assert.equal(details_page.notification_message, "Your review was successfully edited")
        details_page.wait_notification_box_not_visible()

        # Go to reviews page and verify
        reviews = details_page.click_all_reviews_button()
>       Assert.equal(reviews.logged_in_users_review.text, mock_review['body'])

tests/desktop/consumer_pages/test_reviews.py:149:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <class 'unittestzero.Assert'>, first = u'AutomaticReviewSeleniumTests1422546752', second = 'AutomaticReviewSeleniumTests1422546789'
msg = None

    @classmethod
    def equal(self, first, second, msg=None):
        """
            Asserts that 2 elements are the same

            :Args:
             - First object to be tested
             - Second object to be tested
             - Message that will be printed if it fails
            """
>       assert first == second, msg
E       AssertionError: None

../../.virtualenvs/marketplace-tests/lib/python2.7/site-packages/unittestzero.py:55: AssertionError

___________________________________________________ TestSearching.test_results_page_items ___________________________________________________

self = <tests.desktop.consumer_pages.test_search.TestSearching instance at 0x1091e63f8>
mozwebqa = <pytest_mozwebqa.pytest_mozwebqa.TestSetup instance at 0x1091e6680>

    @pytest.mark.sanity
    @pytest.mark.nondestructive
    def test_results_page_items(self, mozwebqa):
        home_page = Home(mozwebqa)
        home_page.go_to_homepage()

        search_term = self._take_first_new_app_name(mozwebqa)
        search_page = home_page.header.search(search_term)

>       search_page.click_expand_button()

tests/desktop/consumer_pages/test_search.py:105:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pages.desktop.consumer_pages.search.Search object at 0x109242d90>

    def click_expand_button(self):
>       self.find_element(*self._expand_button_locator).click()

pages/desktop/consumer_pages/search.py:45:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <pages.desktop.consumer_pages.search.Search object at 0x109242d90>, locator = ('css selector', '#search-results .expand-toggle')

    def find_element(self, *locator):
>       return self._selenium_root.find_element(*locator)

pages/page.py:137:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.firefox.webdriver.WebDriver object at 0x109242b50>, by = 'css selector', value = '#search-results .expand-toggle'

    def find_element(self, by=By.ID, value=None):
        """
            'Private' method used by the find_element_by_* methods.

            :Usage:
                Use the corresponding find_element_by_* instead of this.

            :rtype: WebElement
            """
        if not By.is_valid(by) or not isinstance(value, str):
            raise InvalidSelectorException("Invalid locator values passed in")

        return self.execute(Command.FIND_ELEMENT,
>                            {'using': by, 'value': value})['value']

../../.virtualenvs/marketplace-tests/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py:662:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.firefox.webdriver.WebDriver object at 0x109242b50>, driver_command = 'findElement'
params = {'sessionId': u'61d53465-a148-7f4d-8956-1fe18f29fe30', 'using': 'css selector', 'value': '#search-results .expand-toggle'}

    def execute(self, driver_command, params=None):
        """
            Sends a command to be executed by a command.CommandExecutor.

            :Args:
             - driver_command: The name of the command to execute as a string.
             - params: A dictionary of named parameters to send with the command.

            :Returns:
              The command's JSON response loaded into a dictionary object.
            """
        if self.session_id is not None:
            if not params:
                params = {'sessionId': self.session_id}
            elif 'sessionId' not in params:
                params['sessionId'] = self.session_id

        params = self._wrap_value(params)
        response = self.command_executor.execute(driver_command, params)
        if response:
>           self.error_handler.check_response(response)

../../.virtualenvs/marketplace-tests/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py:173:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x1092427d0>
response = {u'name': u'findElement', u'sessionId': u'61d53465-a148-7f4d-8956-1fe18f29fe30', u'status': 7, u'value': {u'message': ...mponents/driver-component.js:548', u'lineNumber': 5, u'methodName': u'fxdriver.Timer.prototype.setTimeout/<.notify'}]}}

    def check_response(self, response):
        """
            Checks that a JSON response from the WebDriver does not have an error.

            :Args:
             - response - The JSON response from the WebDriver server as a dictionary
               object.

            :Raises: If the response contains an error message.
            """
        status = response['status']
        if status == ErrorCode.SUCCESS:
            return
        exception_class = ErrorInResponseException
        if status in ErrorCode.NO_SUCH_ELEMENT:
            exception_class = NoSuchElementException
        elif status in ErrorCode.NO_SUCH_FRAME:
            exception_class = NoSuchFrameException
        elif status in ErrorCode.NO_SUCH_WINDOW:
            exception_class = NoSuchWindowException
        elif status in ErrorCode.STALE_ELEMENT_REFERENCE:
            exception_class = StaleElementReferenceException
        elif status in ErrorCode.ELEMENT_NOT_VISIBLE:
            exception_class = ElementNotVisibleException
        elif status in ErrorCode.INVALID_ELEMENT_STATE:
            exception_class = InvalidElementStateException
        elif status in ErrorCode.INVALID_SELECTOR \
                or status in ErrorCode.INVALID_XPATH_SELECTOR \
                or status in ErrorCode.INVALID_XPATH_SELECTOR_RETURN_TYPER:
            exception_class = InvalidSelectorException
        elif status in ErrorCode.ELEMENT_IS_NOT_SELECTABLE:
            exception_class = ElementNotSelectableException
        elif status in ErrorCode.INVALID_COOKIE_DOMAIN:
            exception_class = WebDriverException
        elif status in ErrorCode.UNABLE_TO_SET_COOKIE:
            exception_class = WebDriverException
        elif status in ErrorCode.TIMEOUT:
            exception_class = TimeoutException
        elif status in ErrorCode.SCRIPT_TIMEOUT:
            exception_class = TimeoutException
        elif status in ErrorCode.UNKNOWN_ERROR:
            exception_class = WebDriverException
        elif status in ErrorCode.UNEXPECTED_ALERT_OPEN:
            exception_class = UnexpectedAlertPresentException
        elif status in ErrorCode.NO_ALERT_OPEN:
            exception_class = NoAlertPresentException
        elif status in ErrorCode.IME_NOT_AVAILABLE:
            exception_class = ImeNotAvailableException
        elif status in ErrorCode.IME_ENGINE_ACTIVATION_FAILED:
            exception_class = ImeActivationFailedException
        elif status in ErrorCode.MOVE_TARGET_OUT_OF_BOUNDS:
            exception_class = MoveTargetOutOfBoundsException
        else:
            exception_class = WebDriverException
        value = response['value']
        if isinstance(value, basestring):
            if exception_class == ErrorInResponseException:
                raise exception_class(response, value)
            raise exception_class(value)
        message = ''
        if 'message' in value:
            message = value['message']

        screen = None
        if 'screen' in value:
            screen = value['screen']

        stacktrace = None
        if 'stackTrace' in value and value['stackTrace']:
            stacktrace = []
            try:
                for frame in value['stackTrace']:
                    line = self._value_or_default(frame, 'lineNumber', '')
                    file = self._value_or_default(frame, 'fileName', '<anonymous>')
                    if line:
                        file = "%s:%s" % (file, line)
                    meth = self._value_or_default(frame, 'methodName', '<anonymous>')
                    if 'className' in frame:
                        meth = "%s.%s" % (frame['className'], meth)
                    msg = "    at %s (%s)"
                    msg = msg % (meth, file)
                    stacktrace.append(msg)
            except TypeError:
                pass
        if exception_class == ErrorInResponseException:
            raise exception_class(response, message)
        elif exception_class == UnexpectedAlertPresentException and 'alert' in value:
            raise exception_class(message, screen, stacktrace, value['alert'].get('text'))
>       raise exception_class(message, screen, stacktrace)
E       NoSuchElementException: Message: Unable to locate element: {"method":"css selector","selector":"#search-results .expand-toggle"}
E       Stacktrace:
E           at FirefoxDriver.prototype.findElementInternal_ (file:///var/folders/sk/1kj23lvn46n7pvr7_xk074wm0000gn/T/tmph7HsE7/extensions/fxdriver@googlecode.com/components/driver-component.js:9641:26)
E           at fxdriver.Timer.prototype.setTimeout/<.notify (file:///var/folders/sk/1kj23lvn46n7pvr7_xk074wm0000gn/T/tmph7HsE7/extensions/fxdriver@googlecode.com/components/driver-component.js:548:5)

../../.virtualenvs/marketplace-tests/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py:166: NoSuchElementException
```

**Comment by AndreiH** [comment]
Thanks @bobsilverberg , I will take a look at those failing tests and also setup some adhoc jobs

**Comment by AndreiH** [comment]
@bobsilverberg <code>test_that_checks_the_editing_of_a_review</code> is failing both on Stage and Dev because https://bugzilla.mozilla.org/show_bug.cgi?id=1126563 and I xfailed the test.
Also, here are 2 adhoc jobs on DEV:
Mobile: http://selenium.qa.mtv2.mozilla.com:8080/view/Marketplace/job/marketplace.DEV.mobile.saucelabs.NEW_UI/
Desktop: http://selenium.qa.mtv2.mozilla.com:8080/job/marketplace.DEV.saucelabs.NEW_UI

The strange thing is that the 3 remaining failures are not reproducible for me, as you can see from the Saucelabs screenshots and the screencast. This seems that it's loading the previous Home page, also the Promo table is not visible, and this is not reproducible locally with automation or manually.

File: pages/desktop/consumer_pages/details.py [hunk 2]
```diff
@@ -118,15 +109,12 @@ def is_privacy_policy_link_visible(self):
...
-        from pages.desktop.consumer_pages.add_review import AddReview
-        return AddReview(self.testsetup)
-
-    def click_edit_review(self):
-        self.scroll_to_element(*self._edit_review_button_locator)
-        self.selenium.find_element(*self._edit_review_button_locator).click()
+    def click_review_button(self, edit_review=False):
+        self.scroll_to_element(*self._review_button_locator)
+        self.selenium.find_element(*self._review_button_locator).click()
+        if edit_review == False:
```
Comment (bobsilverberg) [inline pages/desktop/consumer_pages/details.py]: This should be `if not edit_review`

File: pages/desktop/consumer_pages/search.py [hunk 3]
```diff
@@ -33,7 +32,6 @@ def __init__(self, testsetup, app_name=False):
             self.app_name = app_name
         else:
             self._page_title = 'Search Results | Firefox Marketplace'
-        self.wait_for_element_present(*self._search_results_section_locator)
```
Comment (bobsilverberg) [inline pages/desktop/consumer_pages/search.py]: This seems a bit dangerous to have removed a wait without finding something to replace it.

File: tests/desktop/consumer_pages/test_home_page.py [hunk 4]
```diff
@@ -65,13 +65,12 @@ def test_opening_every_category_page_from_categories_menu(self, mozwebqa):
             Assert.true(len(category_page.apps) > 0)
             Assert.true(category_page.is_new_popular_tabs_visible)
             Assert.true(category_page.popular_tab_class == 'active')
-            Assert.true(category_page.is_view_all_link_visible)
```
Comment (bobsilverberg) [inline tests/desktop/consumer_pages/test_home_page.py]: Does this link no longer exist? Is there something else we should be checking?

File: tests/desktop/consumer_pages/test_home_page.py [hunk 5]
```diff
@@ -103,7 +102,6 @@ def test_settings_dropdown_menu(self, mozwebqa):
         user_settings = home_page.header.click_edit_account_settings()
         Assert.true(user_settings.is_the_current_page)
         Assert.true(user_settings.is_email_visible)
-        Assert.true(user_settings.is_email_non_editable)
```
Comment (bobsilverberg) [inline tests/desktop/consumer_pages/test_home_page.py]: Is it valid to have removed this assert?

**Comment by bobsilverberg** [comment]
The tests are all passing for me locally, except for a couple of review tests which I can see are actually generating an error from the site (i.e., "There was a problem saving your review"). If that persists, that represents an actual bug, so I'm happy to merge the tests now, and address that bug next.

There is also some weirdness going on with both our local grid and Saucelabs seeming to see a different version of the site which is causing some tests to fail on CI but not locally. This is a mystery that needs to be solved, but shouldn't preclude merging this.

There are a lot of changes here that are needed to restore coverage, so I am going to merge this and fix the one nit I feel needs to be addressed, and then we can start to address the few tests that may still be failing. This will also allow us to proceed with PR #609.

File: tests/desktop/consumer_pages/test_home_page.py (hunk 5 above)
Comment (AndreiH) [inline tests/desktop/consumer_pages/test_home_page.py]: The email is not an input field anymore where we had a class == noneditable. The email is now just text.

File: tests/desktop/consumer_pages/test_home_page.py (hunk 4 above)
Comment (AndreiH) [inline tests/desktop/consumer_pages/test_home_page.py]: This no longer exist to view all apps, there are already shown.

**Comment by AndreiH** [comment]
@bobsilverberg I found out that the 3 remaining failures on Saucelabs are caused because the screen resolution. I could reproduce the exact same failures when I change my VMs resolution to a minimum.
Also the Promo Box disappears .

**Comment by bobsilverberg** [comment]
Thanks @AndreiH. We thought that was the issue too, but I tried changing the screen resolution for a sauce labs job and it didn't fix it. Maybe my command for the screen resolution didn't actually work.

**Comment by AndreiH** [comment]
@bobsilverberg I talked to @davehunt and it turns out that the command was ineffective, it was in json, but the plugin changed some time back to be capability:value instead. Issue Fixed now, builds are green :)

**Pull request merged**